tests/integration/common.rs

```diff
@@ -1,13 +1,158 @@
+use std::path::PathBuf;
+use std::sync::Arc;
+
+use derive_builder::Builder;
+use derive_more::{Deref, DerefMut};
+use sc_network_common::config::MultiaddrWithPeerId;
+use subspace_sdk::farmer::{CacheDescription, PlotDescription};
+use subspace_sdk::node::{chain_spec, ChainSpec, DsnBuilder, NetworkBuilder, Role};
+use tempfile::TempDir;
+
 pub fn setup() {
     let _ = tracing_subscriber::fmt()
         .with_env_filter(
             "debug,parity-db=info,cranelift_codegen=info,wasmtime_cranelift=info,\
              subspace_sdk=trace,subspace_farmer=trace,subspace_service=trace,\
              subspace_farmer::utils::parity_db_store=debug,trie-cache=info,wasm_overrides=info,\
-             libp2p_gossipsub::behaviour=info,wasmtime_jit=info,wasm-runtime=info"
+             jsonrpsee_core=info,libp2p_gossipsub::behaviour=info,wasmtime_jit=info,\
+             wasm-runtime=info"
                 .parse::<tracing_subscriber::EnvFilter>()
                 .expect("Env filter directives are correct"),
         )
         .with_test_writer()
         .try_init();
 }
+
+#[derive(Builder)]
+#[builder(pattern = "immutable", build_fn(private, name = "_build"), name = "NodeBuilder")]
+pub struct InnerNode {
+    #[builder(default)]
+    not_force_synced: bool,
+    #[builder(default)]
+    boot_nodes: Vec<MultiaddrWithPeerId>,
+    #[builder(default)]
+    dsn_boot_nodes: Vec<MultiaddrWithPeerId>,
+    #[builder(default)]
+    not_authority: bool,
```

Comment:
same with `not_authority`

Reply:
Same here, by default test node should be authority to produce blocks, but sometimes we don't need that therefore, we want to disable it